nupic/research/frameworks/pytorch/models/common_models.py

```diff
@@ -49,7 +49,7 @@ def forward(self, x):
 
 class OmniglotCNN(nn.Module):
 
-    def __init__(self, num_classes, **kwargs):
+    def __init__(self, num_classes):
```

Comment:
Now I see why you removed `input_size` from the experiments file. Accepting **kwargs make it easier to switch back and forth between networks without having to change model_args as well. 

I am not opposed to it, but it is an experiments framework, and I think as such we should do everything that makes it easier for users to experiment it with. Having consistent APIs for common classes, like networks, is one of these things that makes it easier to perform experiments (for example I can grid search between 3 different networks).

Reply:
It seems odd to include unused arguments. I agree that experimenting should be made easy, but the arguments should reflect the scope and intention of the class.

I'd argue that maybe this network shouldn't be in frameworks. If there are still iterations to make on the network structure, then maybe this should be in projects.

Reply:
What I believe to be a better approach is instead of throwing yet another error, let it run and just warn the user: "Arguments x, y and z are not being used in this network". The user is probably aware of it already.

I would agree there is probably better solutions, although I can't come up with one now. I just don't think making it even harder to run experiments is a good idea.